src/actions/download.js

```diff
@@ -1,25 +1,13 @@
 import * as types from '../constants/actionTypes.js'
 
-// Set download state (download dialog is open)
-export const toggleDownloadDialog = (show) => ({
-    type: types.DOWNLOAD_DIALOG_TOGGLE,
-    payload: show,
+// Toggle download mode
+export const toggleDownloadMode = (payload) => ({
+    type: types.DOWNLOAD_MODE_TOGGLE,
+    payload,
 })
 
-// Whether the name should be included on the map download
-export const toggleDownloadShowName = (show) => ({
-    type: types.DOWNLOAD_NAME_SHOW_TOGGLE,
-    payload: show,
-})
-
-// Whether the legend should be included on the map download
-export const toggleDownloadShowLegend = (show) => ({
-    type: types.DOWNLOAD_LEGEND_SHOW_TOGGLE,
-    payload: show,
-})
-
-// Legend corner position
-export const setDownloadLegendPosition = (position) => ({
-    type: types.DOWNLOAD_LEGEND_POSITION_SET,
-    payload: position,
+// Set downlond property
+export const setDownloadProperty = (payload) => ({
```

Comment:
What about renaming this to `setDownloadConfig`? It's not clear what "property" means.

Reply:
Renamed to `setDowloadConfig` and `DOWNLOAD_CONFIG_SET`.